neuromation/cli/click_types.py

```diff
@@ -193,3 +194,41 @@ async def async_complete(
 
 
 PRESET = PresetType()
+
+
+class JobType(AsyncType[str]):
+    name = "job"
+
+    async def async_convert(
+        self,
+        root: Root,
+        value: str,
+        param: Optional[click.Parameter],
+        ctx: Optional[click.Context],
+    ) -> str:
+        return value
+
+    async def async_complete(
+        self, root: Root, ctx: click.Context, args: Sequence[str], incomplete: str
+    ) -> List[Tuple[str, Optional[str]]]:
+        async with await root.init_client() as client:
+            ret: List[Tuple[str, Optional[str]]] = []
+            now = datetime.now()
+            async for job in client.jobs.list(
+                since=now - timedelta(days=7), reverse=True, limit=100
+            ):
+                job_name = job.name or ""
+                for test in (
+                    job.id,
+                    job_name,
+                    f"job:{job.id}",
+                    f"job:/{client.username}/{job.id}",
```

Comment:
If you use the form `job:/{user-name}/{job-id}`, `user-name` is usually the other user's name. You can use this name for filtering.

If you use the form `job:{job-id}` (without `/` after `job:`), you refer to your own jobs, so you can pass `owners=[client.username]` to `list()`.

Reply:
thanks, now I take cluster and user from job-description.

Regarding `owners=[client.username]` -- I prefer to make the single REST call for the whole autocomplete, it makes code much simpler. Do you have an objection?

Reply:
Ok, it works now. Maybe adding additional parameters taken from `incomplete` could make the request faster for admins, but I am not sure the difference will be significant.

But I think that you should return a name or an URI when autocomplete a name or an URI.

Reply:
Please ignore the last phrase. I missed that the current code does exactly this.